### DISCUSSION TITLE
Case-sensitive column names

### DISCUSSION THREAD

**Discussion by krlmlr** [duckdb/duckdb#14682] [Ideas]
In R, column names are case-insensitive; duckdb seems to be merely case-preserving (and only if an option is set).

How difficult would it be to optionally support case-sensitive column names, perhaps only for the relational API?

Example: I want the following to return columns `n` and `N`, currently, the second column is returned as `N_1`.

```sh
✗ echo 'with x as (select 1 as a, 1 as n union select 1 as a, 2 as n), y as (select *, sum(n) over(partition by a) as N from x order by a, n) select * from y' | duckdb
```

```
┌───────┬───────┬────────┐
│   a   │   n   │  N_1   │
│ int32 │ int32 │ int128 │
├───────┼───────┼────────┤
│     1 │     1 │      3 │
│     1 │     2 │      3 │
└───────┴───────┴────────┘
```

**Comment by owl2** [comment]
I don't know if it can be considered as a solution but this request have what you need : 
```sql
with x as (
	select 1 as a, 
	1 as "n"
	
	union 
	
	select 1 as "a", 
	2 as "n"
)

select *, 
	sum("n") over(partition by a) as "N"
from x order by a, "n"
```


a|n|N|
-+-+-+
1|1|3|
1|2|3|

**Reply by krlmlr** [reply]
Thanks. The somewhat useless CTE wrapper triggers column name disambiguation. I see how duplicate column names are preserved in the output (as in your example), but they are disambiguated as soon as a result is used (even with a trivial CTE).

I like the disambiguation. I don't like that `n` and `N` are forcibly treated as identical column names, I'm asking for an option to treat those as different column names so that my example works.
